app.py

```diff
@@ -231,24 +232,29 @@ def charge():
     bundles = get_bundles('charge')
 
     if form.validate():
-        customer = stripe.Customer.retrieve(request.form['customerId'])
-        add_customer_and_charge.delay(form=request.form,
-                customer=customer)
-        print('Validated form of customer {} {} {}'.format(customer_email,
-            customer_first, customer_last))
-        if request.form['installment_period'] == 'None':
-            gtm['event_label'] = 'once'
-        else:
-            gtm['event_label'] = request.form['installment_period']
-        gtm['event_value'] = request.form['amount']
-        return render_template('charge.html',
-                amount=request.form['amount'], gtm=gtm, bundles=bundles)
+        """
+        This logic is in a try/except block because rogue IPs have
+        tried to POST to this endpoint with customer IDs that don't exist.
+        """
+        try:
+            customer = stripe.Customer.retrieve(request.form['customerId'])
+            add_customer_and_charge.delay(form=request.form, customer=customer)
+            print('Validated form of customer {} {} {}'.format(customer_email,
+                customer_first, customer_last))
+            if request.form['installment_period'] == 'None':
+                gtm['event_label'] = 'once'
+            else:
+                gtm['event_label'] = request.form['installment_period']
+            gtm['event_value'] = request.form['amount']
+            return render_template('charge.html',
+                    amount=request.form['amount'], gtm=gtm, bundles=bundles)
+        except stripe.error.InvalidRequestError:
```

Comment:
`InvalidRequestError` can be raised for other reasons. So I'd introspect the exception to see if it contains `No such customer:`. Otherwise, if that exception is raised for some other reason it'll be difficult to debug.